src/source/amqp.rs

```diff
@@ -0,0 +1,236 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#![cfg(not(tarpaulin_include))]
+
+use crate::source::prelude::*;
+use serde::{Deserialize};
+use crate::url::TremorUrl;
+use url::Url;
+use crate::errors::Error;
+use lapin::{
+    options::*, types::FieldTable, Connection, ConnectionProperties, Consumer,
+};
+
+/*enum QueueProperties {
+    Durable 1
+    Exclusive 2
+    AutoDelete 4
+    NoWait 8
+}*/
+#[derive(Deserialize, Debug, Clone)]
+#[allow(clippy::struct_excessive_bools)]
+pub struct Config {
+    pub amqp_addr: String,
+    queue_name: String,
+    queue_options: QueueDeclareOptions,
+    routing_key: String,
+    exchange: String,  // "", by default
+    #[serde(default = "Default::default")]
+    pub close_on_done: bool,
+    #[serde(default = "Default::default")]
+    pub sleep_on_done: u64,
```

Comment:
see the nats src, I think the same applies to this config

Reply:
as above, removed these two residuals